### ISSUE TITLE
ASM Module failure 

### ISSUE THREAD

**Issue by uRhos** [terraform-google-modules/terraform-google-kubernetes-engine#1264]
### TL;DR

I was creating a module for a GKE cluster that will set up Anthos and some other things as well, everything runs well and works but the terraform apply fails with this `Error: local-exec provisioner error` on `asm.module.cpr.module.gcloud_kubectl.null_resource.additional_components[0]`

### Expected behavior

Terraform apply gets completed successfully 

### Observed behavior

Terraform apply fails with this error:
`Error: local-exec provisioner error
with module.rho-gke.module.asm.module.cpr.module.gcloud_kubectl.null_resource.additional_components[0]
on .terraform/modules/rho-gke.asm.cpr/main.tf line 174, in resource "null_resource" "additional_components":
  provisioner "local-exec" {
Error running command '.terraform/modules/rho-gke.asm.cpr/scripts/check_components.sh gcloud kubectl': exit status 127. Output: `

### Terraform Configuration

```hcl
provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

data "google_client_config" "default" {}

module "asm" {
  source = "terraform-google-modules/kubernetes-engine/google//modules/asm"

  project_id       = var.project_id
  cluster_name     = module.gke.name
  cluster_location = module.gke.location
}

module "hub" {
  source = "terraform-google-modules/kubernetes-engine/google//modules/fleet-membership"

  project_id   = var.project_id
  cluster_name = module.gke.name
  location     = module.gke.location
}

module "gke" {
  source = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"

  project_id                 = var.project_id
  region                     = var.region
  zones                      = var.zones
  name                       = var.cluster_name
  network                    = var.network
  subnetwork                 = var.subnet
  ip_range_pods              = ""
  ip_range_services          = ""
  http_load_balancing        = true
  gce_pd_csi_driver          = true
  grant_registry_access      = true
  horizontal_pod_autoscaling = true
  network_policy             = true

  enable_private_endpoint = var.enable_private_endpoint
  enable_private_nodes    = true

  master_authorized_networks = var.master_authorized_networks

  create_service_account = true
  node_metadata          = "GKE_METADATA_SERVER"
  config_connector       = true

  istio = false

  maintenance_start_time = var.maintenance_start_time
  maintenance_end_time   = var.maintenance_end_time
  maintenance_recurrence = var.maintenance_recurrence

  kubernetes_version = var.master_version
  release_channel    = var.release_channel

  logging_enabled_components    = []
  monitoring_enabled_components = []

  # Masters CIDR
  master_ipv4_cidr_block = var.master_ipv4_cidr

  remove_default_node_pool = true
  node_pools = [
    {
      name               = "default"
      machine_type       = "n2-standard-4"
      min_count          = 0
      max_count          = 5
      initial_node_count = 0
      disk_size_gb       = 50
      disk_type          = "pd-standard"
      image_type         = "COS_CONTAINERD"
      auto_repair        = true
      auto_upgrade       = true
      autoscaling        = true
      preemptible        = false
    },
    {
      name               = "app-pool"
      machine_type       = var.machine_type
      min_count          = var.min_count
      max_count          = var.max_count
      initial_node_count = var.initial_count
      disk_size_gb       = var.disk_size
      disk_type          = var.disk_type
      image_type         = var.image_type
      auto_repair        = true
      auto_upgrade       = true
      autoscaling        = true
      preemptible        = var.preemptible
    },
    {
      name               = "system"
      machine_type       = var.system_machine_type
      min_count          = var.system_min_count
      max_count          = var.system_max_count
      initial_node_count = var.system_initial_count
      disk_size_gb       = var.system_disk_size
      disk_type          = var.system_disk_type
      image_type         = var.system_image_type
      auto_repair        = true
      auto_upgrade       = true
      autoscaling        = true
      preemptible        = var.system_preemptible
    },
  ]
  node_pools_oauth_scopes = {
    app-pool = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
    system = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
  }

  node_pools_tags = {
    app-pool = [
      "${var.cluster_name}-app-pool",
    ]
    system = [
      "${var.cluster_name}-system",
    ]
  }

  node_pools_labels = {
    app-pool = {
      app-pool = "true"
    }
    system = {
      system = "true"
    }
  }

  node_pools_taints = {
    app-pool = [
      {
        key    = "app-pool"
        value  = "true"
        effect = "NO_SCHEDULE"
      },
    ]
    system = [
      {
        key    = "system"
        value  = "true"
        effect = "NO_SCHEDULE"
      },
    ]
  }
}
```

### Terraform Version

```sh
1.1.9
```

**Comment by shawlz** [comment]
I have this same issue. Any way around this?

**Comment by shawlz** [comment]
@uros-rho Found this issue is because `kubectl` and `gcloud` are not installed. 

```
module "gcloud" {
  source        = "registry.terraform.io/terraform-google-modules/gcloud/google"
  version       = "~> 2.0"
  skip_download = "false"

  platform              = "linux"
  additional_components = ["kubectl", "beta"]

  create_cmd_entrypoint  = "gcloud"
  create_cmd_body        = "version"
  destroy_cmd_entrypoint = "gcloud"
  destroy_cmd_body       = "version"
}
```
Added it using the module above and now getting another error 
![image](https://user-images.githubusercontent.com/222459/171835497-75d8ed8e-e254-47ce-a184-887cc8ecf820.png)

Still trying to figure this out.

**Comment by Felipemirand0** [comment]
any update ?
